lib/pbench/server/api/resources/datasets_daterange.py

```diff
@@ -0,0 +1,72 @@
+from logging import DEBUG, Logger
+
+from flask.json import jsonify
+from flask.wrappers import Request, Response
+from sqlalchemy import func
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    ApiBase,
+    API_OPERATION,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.database.database import Database
+from pbench.server.database.models.datasets import Dataset
+
+
+class DatasetsDaterange(ApiBase):
+    """
+    API class to retrieve the available date range of accessible datasets.
+    """
+
+    GET_SCHEMA = Schema(
+        Parameter("owner", ParamType.USER, required=False),
+        Parameter("access", ParamType.ACCESS, required=False),
+    )
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            None,
+            role=API_OPERATION.READ,
+        )
+
+    def _get(self, _, request: Request) -> Response:
+        """
+        Get the date range for which datasets are available to the client based
+        on authentication plus optional dataset owner and access criteria.
+
+        Args:
+            json_data: For a GET, this contains only the Flask URI template
+                values, which aren't used here
+            request: The original Request object containing query parameters
+
+
+        GET /api/v1/datasets/daterange?owner=user&access=public
+        """
+
+        json = self._collect_query_params(request, self.GET_SCHEMA)
+        new_json = self.GET_SCHEMA.validate(json) if json else json
+
+        access = new_json.get("access")
+        owner = json.get("owner")
```

Comment:
Because `json` is the text user input; `new_json` is the internal representation. In particular, `new_json["owner"]` is the **user ID** rather than the **username**, and for authorization we need the latter.

Reply:
Why are we not using the user ID for authorization?  Shouldn't we be translating the name to an ID before hand?

Reply:
Not that you have to fix that in this PR, but this is a good question to consider.

Reply:
That's an old decision I'm definitely *not* going to re-open in this PR.

I'm going to suggest we defer thinking about this until we figure out exactly how we're integrating with Keycloak which may have some opinions on the subject. 👀

Reply:
It does look odd to be pulling `owner` from `json` and `access` from `new_json`.

(If this really is the right thing to do, perhaps move the assignment of `owner` up above the assignment of `new_json`?)

Reply:
I don't even have a response for all this. I'm just accommodating an existing design, and there are other calls.

But I guess I'm going to put this into draft and go off and spend a day converting `_check_authorization` and all the callers to use ID so all this cruft becomes irrelevant. Ugh.